src/ngraph/runtime/cpu/pass/cpu_layout.cpp

```diff
@@ -1431,6 +1432,34 @@ namespace ngraph
                     }
                 }
 
+                template <>
+                void CPULayout::LAYOUT_DECL(ngraph::op::Slice)
+                {
+                    if (mkldnn_utils::use_mkldnn_kernel(node.get()))
+                    {
+                        // pass input format to output
+                        auto input_md = mkldnn_utils::get_input_mkldnn_md(node.get(), 0);
+                        NGRAPH_DEBUG << "input memory format: " << input_md.data.format << "\n";
+                        auto result_format =
+                            static_cast<mkldnn::memory::format>(input_md.data.format);
+                        if (result_format == mkldnn::memory::blocked)
+                        {
+                            //TODO get named blocked format for output?
```

Comment:
could we try `get_named_md` which might be able to canonicalize a blocked layout into `nChw8c` or `nChw16c` simple_reorder seems to support both.

Reply:
Can we create an MKLDNN pd and query for output format?

Reply:
Output gets its format from input. I have modified the code to call create_blocked_mkldnn_md when input format is blocked.